addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/handler/ZoneMinderThingMonitorHandler.java

```diff
@@ -624,265 +739,451 @@ public void updateStatus(ThingStatus status) {
+                            zmid.getMessage(), zmid.getCause());
+
+                } catch (ZoneMinderGeneralException | ZoneMinderAuthenticationException zme) {
+                    logger.error(
+                            "{}: context='fetchMonitorDaemonStatus' error in call to 'getCaptureDaemonStatus' - Exception='{}', Message={}' ",
+                            getLogIdentifier(), zme.getClass().getCanonicalName(), zme.getMessage(), zme.getCause());
+                } finally {
+                    if (captureDaemon != null) {
+                        dataConverter.setMonitorCaptureDaemonStatus(stateCapture);
+                    }
+
+                }
             }
 
-        } catch (Exception ex) {
-            logger.debug("{}", ex.getMessage());
-        }
+            if (isLinked(ZoneMinderConstants.CHANNEL_MONITOR_ANALYSIS_DAEMON_STATE)) {
+                try {
+                    stateAnalysis = UnDefType.UNDEF;
+                    if (fetchAnalysisFrame) {
+                        analysisDaemon = monitorProxy.getAnalysisDaemonStatus();
+                        logger.debug(
+                                "{}: context='onFetchData' tag='analysisDaemon' URL='{}' ResponseCode='{}' ResponseMessage='{}'",
+                                getLogIdentifier(), analysisDaemon.getHttpRequestUrl(), analysisDaemon.getHttpStatus(),
+                                analysisDaemon.getHttpResponseMessage());
+
+                        stateAnalysis = (analysisDaemon.getStatus() ? OnOffType.ON : OnOffType.OFF);
+                        fetchFrame = true;
+                    }
 
-        return state;
+                } catch (ZoneMinderResponseException zmre) {
+                    logger.error(
+                            "{}: context='fetchMonitorDaemonStatus' error in call to 'getAnalysisDaemonStatus' - Http: Status='{}', Message='{}', ExceptionMessage='{}', Exception='{}'",
+                            getLogIdentifier(), zmre.getHttpStatus(), zmre.getHttpMessage(), zmre.getExceptionMessage(),
+                            zmre.getClass().getCanonicalName(), zmre.getCause());
+                } catch (ZoneMinderInvalidData zmid) {
+                    logger.error(
+                            "{}: context='fetchMonitorDaemonStatus' error in call to 'getAnalysisDaemonStatus' - Response='{}', Exception='{}'",
+                            getLogIdentifier(), zmid.getResponseString(), zmid.getClass().getCanonicalName(),
+                            zmid.getCause());
+
+                } catch (ZoneMinderGeneralException | ZoneMinderAuthenticationException zme) {
+                    logger.error(
+                            "{}: context='fetchMonitorDaemonStatus' error in call to 'getAnalysisDaemonStatus' - Exception='{}' ",
+                            getLogIdentifier(), zme.getClass().getCanonicalName(), zme.getCause());
+                } catch (Exception ex) {
```

Comment:
Please catch specific exceptions instead of catching all exceptions. That way it is more clear what exceptions need to be handled and how this is done.